File: Data Structures & Algorithms/concatenation-of-array/submission-0.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int n = nums.size() * 2;
        vector<int> ans;
        int k = 0;
        for (int i = 0; i < n; i++) {

            if (k == nums.size()) {
                k = 0;
            }
            ans.push_back(nums[k]);
            k++;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {

        int l = 0;
        int r = nums.length - 1;
        int min = 0;
        while (l < r) {
            min = l + (r - l) / 2;

            if (nums[min] > nums[r]) {
                l = min + 1;
            } else {
                r = min;
            }
        }

        return nums[l];
    }
}
